py_src/ipyelk/nx/pipes.py

```diff
@@ -0,0 +1,193 @@
+from typing import Dict, Hashable, Iterator, Optional
+
+import ipywidgets as W
+import networkx as nx
+import traitlets as T
+
+from ipyelk.elements import Edge, HierarchicalElement, Label, Node, Port, Registry
+from ipyelk.elements.serialization import build_edge, build_shape_map, iter_elements
+
+from .. import diagram
+from ..elements import layout_options as opt
+from ..pipes import BrowserTextSizer, ElkJS, Pipe, Pipeline
+
+root_opts = opt.OptionsWidget(
+    options=[
+        opt.HierarchyHandling(),
+    ],
+).value
+label_opts = opt.OptionsWidget(
+    options=[opt.NodeLabelPlacement(horizontal="center", vertical="center")]
+).value
+node_opts = opt.OptionsWidget(
+    options=[
+        opt.NodeSizeConstraints(),
+    ],
+).value
+
+
+class NXSource(W.Widget):
+    graph: nx.MultiDiGraph = T.Instance(nx.MultiDiGraph)
+    hierarchy: nx.DiGraph = T.Instance(nx.DiGraph, allow_none=True)
+
+
+class NetworkxPipe(Pipe):
+    source: NXSource = T.Instance(NXSource, allow_none=True)
+
+    async def run(self):
+        """Run method"""
+        # the things to make the stuff
+        # self.value = elements
+        graph = self.source.graph
+        hierarchy = process_hierarchy(graph, self.source.hierarchy)
+        root: Node = get_root(hierarchy)
+        if not root.layoutOptions:
+            root.layoutOptions = root_opts
+
+        nodes = []
```

Comment:
maybe a list comprehension first and then the fix to the layout options and the labels?

```python
node_elements = [
    from_nx_node(node, data)
    for node, data in graph.nodes(data=True)
]
for element in node_elements:
    if not element.layoutOptions:
        element.layoutOptions = node_opts
    if not element.labels:
        element.labels.append(Label(text=el.id, layoutOptions=label_opts))
```